add-on/src/lib/ipfs-path.js

```diff
@@ -4,10 +4,21 @@
 const IsIpfs = require('is-ipfs')
 
 function safeIpfsPath (urlOrPath) {
+  if (IsIpfs.subdomain(urlOrPath)) {
+    urlOrPath = subdomainToIpfsPath(urlOrPath)
+  }
   // better safe than sorry: https://github.com/ipfs/ipfs-companion/issues/303
   return decodeURIComponent(urlOrPath.replace(/^.*(\/ip(f|n)s\/.+)$/, '$1'))
 }
 
+function subdomainToIpfsPath (urlString) {
+  const url = new URL(urlString)
+  const fqdn = url.hostname.split('.')
+  const cid = fqdn[0]
+  const protocol = fqdn[1]
```

Comment:
Just realized this won't produce correct results for gateways under multi-leveled domains such as `cid.ipfs.something.co.uk`.
Easy fix is to switch to `is-ipfs`, probably something like:

```js
const { subdomainPattern } = require('is-ipfs')

const match = fqdn.match(subdomainPattern)
if (!match) throw new Error('no match for isIPFS.subdomainPattern')
const protocol = match[1]
const cid = match[2]
```